src/SFML/Audio/SoundFileReaderWav.cpp

```diff
@@ -145,7 +146,7 @@ Uint64 SoundFileReaderWav::read(Int16* samples, Uint64 maxCount)
     assert(m_stream);
 
     Uint64 count = 0;
-    while (count < maxCount)
+    while (count < maxCount && m_stream->tell() < m_dataEnd)
```

Comment:
According to the coding style guide, each side of the && expression should be surrounded with ( ).